### PR TITLE
Further adjust tolerance of Octave unit tests

### PR DESCRIPTION
**Pull request by JohannesPfeifer**

### CODE REVIEW

File: tests/observation_trends_and_prefiltering/Trend_diagnostics_MCMC_common.inc [hunk 1]
```diff
@@ -16,10 +16,18 @@ if mean(abs(oo_.FilteredVariables.Mean.Y_obs(1:end-1)-Y_obs(2:end)'))>1e-3 ||...
...
 end
 
-if abs(corr(oo_.FilteredVariables.Mean.Y_obs(2:end-1)-Y_obs(3:end)',oo_.FilteredVariables.Mean.Y_obs(1:end-2)-Y_obs(2:end-1)'))>2e-2 ||...
-    abs(corr(oo_.FilteredVariables.Mean.P_obs(2:end-1)-P_obs(3:end)',oo_.FilteredVariables.Mean.P_obs(1:end-2)-P_obs(2:end-1)'))>2e-2 ||...
-    abs(corr(oo_.FilteredVariables.Mean.junk2(2:end-1)-junk2(3:end)',oo_.FilteredVariables.Mean.junk2(1:end-2)-junk2(2:end-1)'))>2e-2 
-    error('Filtered Variables are wrong')
+if isoctave
+  if abs(corr(oo_.FilteredVariables.Mean.Y_obs(2:end-1)-Y_obs(3:end)',oo_.FilteredVariables.Mean.Y_obs(1:end-2)-Y_obs(2:end-1)'))>2e-2 ||...
+      abs(corr(oo_.FilteredVariables.Mean.P_obs(2:end-1)-P_obs(3:end)',oo_.FilteredVariables.Mean.P_obs(1:end-2)-P_obs(2:end-1)'))>2e-2 ||...
+      abs(corr(oo_.FilteredVariables.Mean.junk2(2:end-1)-junk2(3:end)',oo_.FilteredVariables.Mean.junk2(1:end-2)-junk2(2:end-1)'))>3e-2
```
Comment (houtanb) [inline tests/observation_trends_and_prefiltering/Trend_diagnostics_MCMC_common.inc]: @JohannesPfeifer Is the only difference between these two sets of conditionals the `3e-2` at the end of this line? If so, can you adopt the same strategy you used on lines `42` - `45`? That might make it easier to decipher the difference in the test.

File: tests/kalman/likelihood_from_dynare/fs2000ns_estimation_check.inc [hunk 2]
```diff
@@ -26,8 +26,14 @@ if max(max(abs(SmoothedShocks-repmat(SmoothedShocks(:,:,1),[1,1,2]))))>1e-8
...
-if max(max(abs(SmoothedVariables-repmat(SmoothedVariables(:,:,1),[1,1,2]))))>1e-8
-    error('SmoothedVariables do not match')
+if isoctave
+  if max(max(abs(SmoothedVariables-repmat(SmoothedVariables(:,:,1),[1,1,2]))))>5e-6
+      error('SmoothedVariables do not match')
+  end
+else
+  if max(max(abs(SmoothedVariables-repmat(SmoothedVariables(:,:,1),[1,1,2]))))>1e-8
+      error('SmoothedVariables do not match')
+  end
```
Comment (houtanb) [inline tests/kalman/likelihood_from_dynare/fs2000ns_estimation_check.inc]: Same comment here as in `tests/observation_trends_and_prefiltering/Trend_diagnostics_MCMC_common.inc`

File: tests/deterministic_simulations/rbc_det_stack_solve_algo_7.mod [hunk 3]
```diff
@@ -95,10 +95,16 @@ rplot Consumption;
...
+  if norm(D.oo_.endo_simul - oo_.endo_simul) > 1e-4;
+     disp(norm(D.oo_.endo_simul - oo_.endo_simul));
+     error(sprintf('rbc_det_stack_solve_algo_7 failed with solve_algo=%u',options_.solve_algo));
+  end;
+else
+  if norm(D.oo_.endo_simul - oo_.endo_simul) > 1e-8;
+     disp(norm(D.oo_.endo_simul - oo_.endo_simul));
+     error(sprintf('rbc_det_stack_solve_algo_7 failed with solve_algo=%u',options_.solve_algo));
+  end;
+end
```
Comment (houtanb) [inline tests/deterministic_simulations/rbc_det_stack_solve_algo_7.mod]: @JohannesPfeifer Same comment here as in `tests/observation_trends_and_prefiltering/Trend_diagnostics_MCMC_common.inc`

File: tests/observation_trends_and_prefiltering/Trend_diagnostics_MCMC_common.inc [hunk 4]
```diff
@@ -30,18 +38,24 @@ if max(abs(squeeze(oo_.FilteredVariablesKStepAhead(1,1,2:end-(options_.nk-1)))-o
     mean(squeeze(oo_.FilteredVariablesKStepAhead(2,3,3:end-options_.nk)))>1e-1 
     error('FilteredVariablesKStepAhead is wrong')
 end
+
+if isoctave
+  tol_crit=7e-4;
+else
+  tol_crit=5e-4;
+end
```
Comment (houtanb) [inline tests/observation_trends_and_prefiltering/Trend_diagnostics_MCMC_common.inc]: @JohannesPfeifer I prefer this solution as it makes it easier to see what differs in the tests.

**Comment by JohannesPfeifer** [comment]
@houtanb I amended the commit according to your suggestions

**Pull request merged**